src/workflow/trinary_input.c

```diff
@@ -50,9 +69,9 @@ bool workflow_trinary_input_wordlist(
             "Restore",
             trinary_input_string_create_wordlist(
                 title, wordlist, wordlist_size, _confirm, NULL, _cancel, NULL))) {
-        return false;
+        return _cancel_reason;
```

Comment:
Why not make `workflow_cancel_run` return `workflow_trinary_input_result_t`? I think it might simplify some parts and remove the need for `_cancel_reason` static var.

Reply:
`workflow_cancel_run()` can be used in any context, so the return would have to be `void*`, and internally in cancel.c passed via `workflow_cancel_force()`. Then there would be a static var in cancel.c instead of here.

Not sure how nice all of that would be vs. a local static var.

Reply:
Oh, I see what you mean now. Need to read a bit more code. :+1: